2_process/src/match_sites_reaches.R

```diff
@@ -0,0 +1,65 @@
+#' original author: David Watkins
+#' from https://code.usgs.gov/wwatkins/national-site-reach/-/blob/master/R/match_sites_reaches.R
+#' modified by: Jeff Sadler
+#' Match each site with a reach (seg_id/COMID)
+get_site_flowlines <- function(reach_sf, sites, sites_crs, max_matches = 1, search_radius = 0.1) {
+  #' 
+  #' @description Function to match reaches (flowlines) to point sites 
+  #'
+  #' @param reach_sf sf object of reach polylines with column "subsegid" and in WGS84
+  #' @param sites dataframe with columns "lat" "lon"
+  #' @param sites_crs the crs os the sites table (i.e., 4269 for NAD83)
+  #' @param max_matches the maximum number of segments that a point can match to
+  #' @param search_radius the maximum radius in same units as sf object
+  #' within which a segment will match (segments outside of the radius will not match)
+  #'
+  #' @value A data frame the same columns as the sites input dataframe with additional columns
+  #' of "segidnat", "subsegid" and "offset" where "offset" is the distance (in degrees) between the point
+  #' and matching flowline
+  
+  # set up NHDPlus fields used by get_flowline_index
+  # Note: that we are renaming the subsegid column to COMID because the nhdplusTools
```

Comment:
Thanks for adding this documentation! I think it will help me to remember how we're using the nhdplusTools functions for the PRMS flowlines.